frontend/device/kobo/device.lua

```diff
@@ -740,6 +750,62 @@ function Kobo:toggleGSensor(toggle)
     end
 end
 
+function Kobo:toggleChargingLED(toggle)
+    if not self:canToggleChargingLED() then
+        return
+    end
+
+    -- We have no way of querying the current state from the HW!
+    if toggle == nil then
+        return
+    end
+
+    -- NOTE: While most/all Kobos actually have a charging LED, and it can usually be fiddled with in a similar fashion,
+    --       we've seen *extremely* weird behavior in the past when playing with it on older devices (c.f., #5479).
+    --       In fact, Nickel itself doesn't provide this feature on said older devices
+    --       (when it does, it's an option in the Energy saving settings),
+    --       which is why we also limit ourselves to "true" Mk. 7 devices.
+    local f = io.open("/sys/devices/platform/ntx_led/lit", "w")
+    if not f then
+        logger.err("cannot open /sys/devices/platform/ntx_led/lit for writing!")
+        return false
+    end
+
+    -- c.f., strace -fittvyy -e trace=ioctl,file,signal,ipc,desc -s 256 -o /tmp/nickel.log -p $(pidof -s nickel) &
+    -- This was observed on a Forma, so I'm mildly hopeful that it's safe on other Mk. 7 devices ;).
+    if toggle == true then
+        -- NOTE: Technically, Nickel forces a toggle off before that, too.
+        --       But since we do that on startup, it shouldn't be necessary here...
+        f:write("ch 4")
+        f:flush()
```

Comment:
Do you need to flush after each write?, if so, wouldn't be more legible to write `f:write("ch4"):flush()`

Reply:
They are individual commands, so, yeah, I assume they really need to be each their own separate write syscall.

(That's certainly how it happens in Nickel).

Reply:
> That doesn't actually work ;p. (io.write doesn't actually return the I/O object)

Duh, sorry for suggesting without testing :p

Reply:
Eh, seemed sensible enough ;).